components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/authcontext/JWTTokenGenerator.java

```diff
@@ -226,6 +227,12 @@ public void generateToken(OAuth2TokenValidationMessageContext messageContext) th
         claimsSetBuilder.claim(API_GATEWAY_ID + "/enduser", authzUser);
         //TODO: check setting audience
 
+        if (messageContext.getProperty(OAuthConstants.ACCESS_TOKEN_DO) != null &&
```

Comment:
We have 
        AccessTokenDO accessTokenDO = (AccessTokenDO) messageContext.getProperty("AccessTokenDO");

at the beginning of the method. Can't we use that variable here?

Reply:
fixed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1459/commits/0d9078223b61023395b613d0e075f1ef2f3845f7